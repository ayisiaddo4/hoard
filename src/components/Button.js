import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, StyleSheet, View } from 'react-native';

export default class Button extends Component {
  static propTypes = {
    style: View.propTypes.style,
    onPress: PropTypes.func.isRequired,
    children: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.node,
      PropTypes.arrayOf(PropTypes.node)
    ]).isRequired
  };

  static defaultProps = {
    style: {}
  };

  render() {
    let buttonContent;
    if (typeof this.props.children === 'string') {
      buttonContent = (
        <Text style={styles.buttonText}>{this.props.children}</Text>
      );
    } else {
      buttonContent = this.props.children;
    }

    return (
      <TouchableOpacity
        style={[styles.buttonContainer, this.props.style]}
        onPress={this.props.onPress}
      >
        {buttonContent}
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    backgroundColor: '#fff',
    paddingVertical: 10,
    marginBottom: 30
  },
  buttonText: {
    textAlign: 'center',
    color: '#223252',
    fontWeight: '700'
  }
});
