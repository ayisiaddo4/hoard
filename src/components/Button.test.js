/* eslint-env jest */

import React from 'react';
import { Text } from 'react-native';
import Button from './Button';

import renderer from 'react-test-renderer';

const onPressEvent = jest.fn();

test('renders correctly', () => {
  const tree = renderer
    .create(
      <Button onPress={onPressEvent}>
        <Text>Test</Text>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
