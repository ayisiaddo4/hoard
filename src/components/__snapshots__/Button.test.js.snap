// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 100,
      "opacity": 1,
      "paddingVertical": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 4,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test
  </Text>
</View>
`;
